### PR TITLE
Feat/test coverage

### PR DESCRIPTION
**Pull request by guru-web3**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

File: tests/controller/mockData.ts [hunk 1]
```diff
@@ -149,7 +149,51 @@ export const mockData = {
             mint_address: "",
           },
         ],
-
+        displayActivities: {
```
Comment (ieow) [inline tests/controller/mockData.ts:152]: Solana backend does not have this data.
displayActivity should be constructed from preference controller which should not be mocked

File: tests/controller/tkey.test.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  });
+  it("setMetadata", async () => {
+    const result = await storageLayer.setMetadata({
+      privKey: new BN(ecc_privateKey),
+      input: openloginFaker[0],
+    });
+    assert.deepEqual(result, { message: "", success: true });
+  });
+  it("getMetadata", async () => {
+    sandbox.stub(tkey, "decrypt").callsFake(async () => {
```
Comment (ieow) [inline tests/controller/tkey.test.ts:57]: Why stub decrypt ?
the nock should had return the mock data

File: tests/controller/util.test.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  });
+  it("generateSPLTransaction", async () => {
+    const result = await solanaHelper.generateSPLTransaction(
+      sKeyPair[1].publicKey.toBase58(),
+      1,
+      mockNFTs[0],
+      sKeyPair[0].publicKey.toBase58(),
+      mockGetConnection()
+    );
+    assert.deepEqual(result.feePayer, sKeyPair[0].publicKey);
```
Comment (ieow) [inline tests/controller/util.test.ts:100]: Could add checking and cases where the receiver is sol account address and token account address

Comment (ieow) [reply tests/controller/util.test.ts:100]: Also case where create Account Instruction is needed

File: tests/controller/util.test.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  });
+  it("decodeAllInstruction", async () => {
+    const message = [
```
Comment (ieow) [inline tests/controller/util.test.ts:103]: We can test the decodeInstruction by using createInstruction

File: tests/controller/util.test.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
+import { LAMPORTS_PER_SOL, SystemProgram, Transaction } from "@solana/web3.js";
```
Comment (ieow) [inline tests/controller/util.test.ts:1]: I think the file name should be solanaHelper.test.ts as we are doing unit test on the file

File: tests/controller/tkey.test.ts (hunk 2 above)
Comment (guru-web3) [reply tests/controller/tkey.test.ts:57]: throwing invalid mac when we are nocking since it is a dynamic value!

Comment (ieow) [reply tests/controller/tkey.test.ts:57]: Can't we setMetadata then getMetadata
We just need to check the data passed to the setMetada is the same as the getMetadata
for Nock, we can have a persistence map that keep the data and return on request rite

**Review by ieow — APPROVED** [review]
lgtm

need to update with PR #376 which might change the Openlogin's UserInfo data structure

**Pull request merged**